bytecode/tests/test_cfg.py

```diff
@@ -429,5 +430,163 @@ def test_get_block_index(self):
+            else:
+                return 1
+
+        self.check_stack_size(test)
+
+    def test_stack_size_computation_while_loop_break(self):
+        def test(arg1, *args, **kwargs):
+            while True:
+                if arg1:
+                    break
+
+        self.check_stack_size(test)
+
+    def test_stack_size_computation_with(self):
+        def test(arg1, *args, **kwargs):
+            with open(arg1) as f:
+                return f.read()
+
+        import dis
+        dis.dis(test.__code__)
+        self.check_stack_size(test)
+
+    def test_stack_size_computation_try_except(self):
+        def test(arg1, *args, **kwargs):
+            try:
+                return args[0]
+            except Exception:
+                return 2
+
+        self.check_stack_size(test)
+
+    def test_stack_size_computation_try_finally(self):
+        def test(arg1, *args, **kwargs):
+            try:
+                return args[0]
+            finally:
+                return 2
+
+        self.check_stack_size(test)
+
+    def test_stack_size_computation_try_except_finally(self):
+        def test(arg1, *args, **kwargs):
+            try:
+                return args[0]
+            except Exception:
+                return 2
+            finally:
+                print('Interrupt')
+
+        self.check_stack_size(test)
+
+    def test_stack_size_computation_try_except_else_finally(self):
+        def test(arg1, *args, **kwargs):
+            try:
+                return args[0]
+            except Exception:
+                return 2
+            else:
+                return arg1
+            finally:
+                print('Interrupt')
+
+        self.check_stack_size(test)
+
+    def test_stack_size_computation_nested_try_except_finally(self):
+        def test(arg1, *args, **kwargs):
+            k = 1
+            try:
+                getattr(arg1, k)
+            except AttributeError:
+                pass
+            except Exception:
+                try:
+                    assert False
+                except Exception:
+                    return 2
+                finally:
+                    print('unexpected')
+            finally:
+                print('attempted to get {}'.format(k))
+
+        self.check_stack_size(test)
+
+    def test_stack_size_computation_nested_try_except_else_finally(self):
+        def test(*args, **kwargs):
+            try:
+                v = args[1]
+            except IndexError:
+                try:
+                    w = kwargs['value']
+                except KeyError:
+                    return -1
+                else:
+                    return w
+                finally:
+                    print('second finally')
+            else:
+                return v
+            finally:
+                print('first finally')
+
+        # A direct comparison of the stack depth fails because CPython
+        # generate dead code that is used in stack computation.
```

Comment:
Hum, I would prefer to test hardcoded values, instead of using bytecode assembler, to only test one think (compute_stacksize), not two (compute+assembler).